legion_airflow/legion_airflow/hooks/s3_hook.py

```diff
@@ -39,6 +42,21 @@ def __init__(self, conn_id: str, *args, **kwargs):
         self.key_prefix = self.extras.get('key_prefix', conf.get('core', 's3_bucket_path'))
         self.bucket_prefix = self.extras.get('bucket_prefix', '')
 
+    @staticmethod
+    def parse_s3_url(s3url):
+        """
+        Parse S3 URL into bucket and key
+        :param s3url: S3 URL
+        :return: (bucket, key)
+        """
+        parsed_url = urlparse(s3url)
```

Comment:
This method does not look like usable outside of the S3Hook context. Looks more like private to me.

Reply:
Fixed